features/FEATURE_BLE/targets/TARGET_CORDIO/source/CordioBLE.cpp

```diff
@@ -389,19 +393,22 @@ void BLE::callDispatcher()
     _event_queue.process();
 
     // follow by stack events
-    static uint32_t lastTimeUs = us_ticker_read();
-    uint32_t currTimeUs, deltaTimeMs;
+    static us_timestamp_t last_time_us = _timer.read_high_resolution_us();
 
     // Update the current cordio time
-    currTimeUs = us_ticker_read();
-    deltaTimeMs = (currTimeUs - lastTimeUs) / 1000;
-    if (deltaTimeMs > 0) {
-        WsfTimerUpdate(deltaTimeMs / WSF_MS_PER_TICK);
-        lastTimeUs += deltaTimeMs * 1000;
+    us_timestamp_t curr_time_us = _timer.read_high_resolution_us();
+    uint64_t delta_time_ms = ((curr_time_us - last_time_us) / 1000);
```

Comment:
is there rollover protection somewhere? don't we have a function to safely compare time?